presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java

```diff
@@ -7061,4 +7061,13 @@ public void testLikePrefixAndSuffix()
         assertQuery("select x like '%a%' from (values 'xa bc', 'xabcy', 'abcd') T(x)");
         assertQuery("select x like '%acd%xy%' from (values 'xa bc', 'xabcy', 'abcd') T(x)");
     }
+
+    @Test
+    public void testLambdaExpressionPullUp()
+    {
+        assertQueryWithSameQueryRunner("select map_filter(idmap, (k, v) -> array_position(array_sort(map_keys(idmap)), k) <= 3) from (values map(array[1, 4, 2, 8, 0], " +
+                "array['a', 'a', 'a', 'a', 'a'])) t(idmap)", "values map(array[0, 1, 2], array['a', 'a', 'a'])");
+        assertQuery("select * from (values (1, 1, array[1, 2, 3]), (1, 2, array[0, 1, 4]), (2, 2, array[1, 2, 3]), (5, 0, array[1, 2, 3]))t(id1, id2, array) where any_match(array, x -> x > id1 + id2)",
+                "values (1, 1, array[1, 2, 3]), (1, 2, array[0, 1, 4])");
+    }
```

Comment:
Add tests for nested lambda (also maybe add the plan as the output so we are sure things are pulled out)

Reply:
Added test for nested lambda

> also maybe add the plan as the output so we are sure things are pulled out

This is checked in unit test TestPullUpExpressionInLambdaRules